### ISSUE TITLE
md5sum unstable on nlinkfs

### ISSUE THREAD

**Issue by callegar** [rene/nlinkfs#2]
I have a weird issue when trying to use md5sum (or sha256sum) on files in a filesystem mounted with nlinkfs.

The checksum seems to change at every invocation:

```
md5sum r1.zip 
f672d008f6aad7de780007bf91146369  r1.zip
md5sum r1.zip 
29e7ef1c96098e56e2ae2ccf1b3b2bd7  r1.zip
```
This is somehow scary...

To make the matter even more strange, if I diff the file with a master diff reports no differences.

**Comment by callegar** [comment]
Some further investigation:
- when you call md5sum, the first invocation appears to return the correct checksum
- subsequent invocations seem to return random values as the checksum
- If you put enough time or other operations in between the tests also the subsequent invocations appear to return the right checksum
- If you use `dd` with no parameters you seem to always get the correct checksum
- But if you use `dd` with a large block size (`bs` parameter) you are back to random values.

If you want to test, I am testing with files as large as 180MB or more.

**Comment by rene** [comment]
@callegar , I could reproduce this bug. Although I wrote this code back to 2010, I will take a look... thanks for reporting....

**Comment by rene** [comment]
Fixed by 6e8486076c98c732fe0e93991b48873a000dec45
